src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/failover/FailoverConnectionPlugin.java

```diff
@@ -410,10 +410,11 @@ private void attemptConnectionUsingCachedTopology() throws SQLException {
 
     metricsContainer.registerUseCachedTopology(true);
 
-    int candidateIndex = getCandidateIndexForInitialConnection();
-    if (candidateIndex != NO_CONNECTION_INDEX) {
-      connectTo(candidateIndex);
-    }
+    // Disabling the following because it creates Access Denied when using AwsSecretsManagerPlugin and has no proven performance benefit
```

Comment:
Can you remove this code?

Reply:
Will do. I'll update the comment  as well to say what the code was

Reply:
I just noticed that this also leaves some unused private methods. I'll remove those as well.

Reply:
Removed commented out code and comments